lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1137,34 +1111,30 @@ public function prepare_item_for_response( $post, $request ) {
 		}
 
 		if ( ! empty( $schema['properties']['title'] ) ) {
+			add_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
 			$data['title'] = array(
 				'raw'      => $post->post_title,
 				'rendered' => get_the_title( $post->ID ),
 			);
+			remove_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
 		}
 
 		if ( ! empty( $schema['properties']['content'] ) ) {
-
-			if ( ! empty( $post->post_password ) ) {
-				$this->prepare_password_response( $post->post_password );
-			}
-
 			$data['content'] = array(
-				'raw'      => $post->post_content,
+				'raw'       => $post->post_content,
 				/** This filter is documented in wp-includes/post-template.php */
-				'rendered' => apply_filters( 'the_content', $post->post_content ),
+				'rendered'  => post_password_required( $post ) ? '' : apply_filters( 'the_content', $post->post_content ),
+				'protected' => (bool) $post->post_password,
```

Comment:
The reason we put it under only these fields is ot make it clear that it's only the `content` and the `excerpt` that is protected. The post it's self is not protected, hence it not being a property of the post object it's self. "protected posts" is a bit of a misnomer, "posts with protected content and excerpt". As for duplication, we'll need a `protected: false` on every post either way. However, it's true that (I think anyway) it's not possible to have an excerpt which is public and content which is private, however I'm not too concerned for the duplication. it's not really any more bytes over the wire with gzipping.